src/main/kotlin/com/terraformation/backend/importer/CsvImporter.kt

```diff
@@ -68,7 +79,7 @@ abstract class CsvImporter(
     uploadStore.requireAwaitingAction(uploadId)
 
     dslContext.transaction { _ ->
-      scheduler.enqueue<CsvImporter> { importCsv(uploadId, overwriteExisting) }
```

Comment:
hoping this isn't a dumb question, would inferring the implementing class type work? something like KClass::class

Reply:
The underlying problem is that the "implementing" class that JobRunr detects is the class that contains the lambda function. The bug was that because the lambda was in `CsvImporter`, when it woke up and tried to run the job, it was trying to find a Spring bean of type `CsvImporter` and bombing out because there were three of them (the three importer classes) and no way to tell which one to use.

The class name in angle brackets is a bit of a red herring. It isn't actually used at runtime at all. Its presence (regardless of which class it specifies) is an input to the compiler's type inference and changes which overload of `enqueue()` ends up getting called. Kind of confusing, I know!